ESP8266-PIR-OLED-Arduino-OTA.ino

```diff
@@ -0,0 +1,339 @@
+//DHT, Motion and OLED sketch for "The Kube" sensor
+
+//by bkpsu, https://www.thingiverse.com/thing:2539897
+
+#define MQTT_SOCKET_TIMEOUT 120
+
+/************ HARDWARE CONFIG (CHANGE THESE FOR YOUR SENSOR SETUP) ******************/
+#define REMOTE //Uncomment to enable remote sensor functionality (Wifi & MQTT)
+//#define OLED_SPI //Uncomment if using SPI OLED screen (assuming I2C otherwise)
+#define CELSIUS //Uncomment if you want temperature displayed in Celsius
+//#define DEEP_SLEEP //Uncomment if you want sensor to sleep after every update (Does NOT work with MOTION_ON or LED_ON which require constant uptime)
+#define FLIP_SCREEN //Uncomment if mounting to wall with USB connector on top
+#define MOTION_ON //Uncomment if using motion sensor
+//#define OLED_MOTION //Uncomment if you want screen to turn on only if motion is detected
+//#define LED_ON //Uncomment if using as LED controller
+//#define PRESS_ON //Uncomment if using as Pressure monitor
+#define OTA_UPDATE //Uncomment if using OTA updates
+
+/************ WIFI, OTA and MQTT INFORMATION (CHANGE THESE FOR YOUR SETUP) ******************/
+#define wifi_ssid "wifi_ssid" //enter your WIFI SSID
+#define wifi_password "wifi_password" //enter your WIFI Password
+#define mqtt_server "mqtt_server" // Enter your MQTT server address or IP.
+#define mqtt_device "mqtt_device" //MQTT device
+#define mqtt_user "" //enter your MQTT username
+#define mqtt_password "" //enter your password
+#define OTApassword "123" // change this to whatever password you want to use when you upload OTA
+#define OTANAME "Esp-pir-oled"
+int OTAport = 8266;
+
+/****************************** MQTT TOPICS (change these topics as you wish)  ***************************************/
+#define temperaturepub "home/mqtt_device/temperature"
+#define humiditypub "home/mqtt_device/humidity"
+#define tempindexpub "home/mqtt_device/temperatureindex"
+#define motionpub "home/mqtt_device/motion"
+#define presspub  "home/mqtt_device/pressure"
+
+/****************************** DHT 22 Calibration settings *************/
+
+float temp_offset = -12.2;
+float hum_offset = 14.9;
+
+/**************************************************/
+
+#define DHTPIN 4      // (D2) what digital pin we're connected to
+#define MOTIONPIN 5   // (D1) what digital pin the motion sensor is connected to
+#define PRESSPIN 5    // (D1) what digital pin the motion sensor is connected to
+
+#define DHTTYPE DHT22   // DHT 22/11 (AM2302), AM2321
+
+#include <PubSubClient.h>
+#include "DHT.h"
+
+#ifdef OLED_SPI
+  #include "SSD1306Spi.h" //OLED Lib for SPI version
+#else
+  #include <Wire.h>  // Only needed for Arduino 1.6.5 and earlier
+  #include "SSD1306.h" //OLED Lib for I2C version
+#endif
+
+#ifdef OTA_UPDATE
+  #include <ESP8266WiFi.h>
```

Comment:
Pull #include <ESP8266WiFi.h> out above the OTA_UPDATE clause - it is used in both cases.